src/platform/PICO/usb/usb_msc_pico.c

```diff
@@ -0,0 +1,204 @@
+
+int32_t tud_msc_write10_cb(uint8_t lun, uint32_t lba, uint32_t offset, uint8_t *buffer, uint32_t bufsize)
+{
+    UNUSED(lun);
+    if (!pico_msc_use_sdcard) return -1;
+    if (offset == 0 && (bufsize % 512u) == 0) {
+        uint16_t block_len = (uint16_t)(bufsize / 512u);
+        for (uint16_t i = 0; i < block_len; i++) {
+            while (sdcard_writeBlock(lba + i, (uint8_t *)buffer + (512u * i), NULL, 0) != SDCARD_OPERATION_IN_PROGRESS) {
+                sdcard_poll();
+            }
+            while (sdcard_poll() == 0) {}
+        }
+        mscSetActive();
+        return (int32_t)bufsize;
+    }
+    // Fallback: read-modify-write block
+    uint8_t temp[512];
+    while (sdcard_readBlock(lba, temp, NULL, 0) == 0) {}
+    while (sdcard_poll() == 0) {}
+    const uint32_t n = MIN(bufsize, 512u - offset);
+    memcpy(temp + offset, buffer, n);
+    while (sdcard_writeBlock(lba, temp, NULL, 0) != SDCARD_OPERATION_IN_PROGRESS) {
+        sdcard_poll();
+    }
+    while (sdcard_poll() == 0) {}
+    mscSetActive();
+    return (int32_t)n;
+}
+
+// Invoked to determine disk size
+void tud_msc_capacity_cb(uint8_t lun, uint32_t *block_count, uint16_t *block_size)
+{
+    UNUSED(lun);
+    uint32_t count = 0;
+    uint32_t size32 = 512;
+    if (storage_get_capacity(&count, &size32) == 0) {
+        *block_count = count;
+        *block_size = (uint16_t)size32;
+    } else {
+        *block_count = 0;
+        *block_size = 512;
+    }
+}
+
+// Invoked to check if device is ready for read/write
+bool tud_msc_test_unit_ready_cb(uint8_t lun)
+{
+    UNUSED(lun);
+    if (!pico_msc_use_sdcard) return false;
+    return sdcard_poll() != 0;
+}
+
+// SCSI INQUIRY
+void tud_msc_inquiry_cb(uint8_t lun, uint8_t vendor_id[8], uint8_t product_id[16], uint8_t product_rev[4])
+{
+    UNUSED(lun);
+    const char vid[8] = { 'B','e','t','a','F','l','t',' ' };
+    const char pid[16] = { 'S','D','C','a','r','d',' ','M','S','C',' ',' ',' ',' ',' ',' ' };
+    const char rev[4] = { '0','.','0','1' };
+    memcpy(vendor_id, vid, sizeof(vid));
+    memcpy(product_id, pid, sizeof(pid));
+    memcpy(product_rev, rev, sizeof(rev));
+}
+
+// SCSI REQUEST SENSE, leave default if not needed
+bool tud_msc_start_stop_cb(uint8_t lun, uint8_t power_condition, bool start, bool load_eject)
+{
+    UNUSED(lun); UNUSED(power_condition); UNUSED(start); UNUSED(load_eject);
+    return true;
+}
+
+// Handle other SCSI commands not handled by simple callbacks above
+int32_t tud_msc_scsi_cb(uint8_t lun, uint8_t const scsi_cmd[16], void* buffer, uint16_t bufsize)
+{
+    UNUSED(lun);
+    UNUSED(scsi_cmd);
+    UNUSED(buffer);
+    UNUSED(bufsize);
+    // Return -1 to indicate unsupported command
+    return -1;
+}
+
+uint8_t mscStart(void)
+{
+    // Decide backend based on configuration; default to SD card if available
+#ifdef USE_SDCARD
+    pico_msc_use_sdcard = true;
+    // Ensure SD card initialized
+    sdcard_init(sdcardConfig());
+    while (sdcard_poll() == 0) {}
```

Comment:
(this will prevent exiting MSC using button when sdcard fails/is not present)

Reply:
`mscStart()` now returns 1 if card is not present.